google/cloud/datastore/query.py

```diff
@@ -209,30 +329,62 @@ def filters(self):
         """
         return self._filters[:]
 
-    def add_filter(self, property_name, operator, value):
+    def add_filter(
+        self,
+        property_name=None,
+        operator=None,
+        value=None,
+        *,
+        composite_filter=None,
+        property_filter=None,
+    ):
         """Filter the query based on a property name, operator and a value.
 
         Expressions take the form of::
 
-          .add_filter('<property>', '<operator>', <value>)
+          .add_filter(
+            property_filter=PropertyFilter('<property>', '<operator>', <value>)
+          )
 
         where property is a property stored on the entity in the datastore
         and operator is one of ``OPERATORS``
         (ie, ``=``, ``<``, ``<=``, ``>``, ``>=``, ``!=``, ``IN``, ``NOT_IN``):
 
+        Both AND and OR operations are supported through the 'composite_filter' parameter::
+
+           .add_filter(
+               composite_filter=And(
+                   [
+                       PropertyFilter('<property>', '<operator>', <value>),
+                       PropertyFilter('<property>', '<operator>', <value>)
+
+                   ]
+               )
+           )
+
+           .add_filter(
+               composite_filter=Or(
+                   [
+                       PropertyFilter('<property>', '<operator>', <value>),
+                       PropertyFilter('<property>', '<operator>', <value>)
+                   ]
+               )
+           )
+
         .. testsetup:: query-filter
 
             import uuid
 
             from google.cloud import datastore
+            from google.cloud.datastore.query import PropertyFilter
 
             client = datastore.Client()
 
         .. doctest:: query-filter
 
             >>> query = client.query(kind='Person')
-            >>> query = query.add_filter('name', '=', 'James')
```

Comment:
what happens to this existing code? is it a warning?

Reply:
Yes there will be a `DeprecationWarning` if they use it. (it still works)
https://github.com/googleapis/python-datastore/pull/418/files#diff-e90ee28143f5bf4b16f8a49b045c7df27c5019c55d0481ef0fd3cb224514385dR407

Reply:
I have changed this to `UserWarning`.